File: Data Structures & Algorithms/top-k-elements-in-list/submission-0.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        
        groups={}

        for num in nums:
            if num not in groups:
                groups[num]=1
            else:
                groups[num]+=1
            
        sorted_dict = {k: v for k, v in sorted(groups.items(), key=lambda item: item[1], reverse=True)}

        freqArr=[]
        
        counter=0
        for key in sorted_dict:
            if counter==k:
                break
            else:
                freqArr.append(key)   
                counter+=1          

        return freqArr

File: Data Structures & Algorithms/find-duplicate-integer/submission-0.cpp
class Solution {
public:
    int findDuplicate(vector<int>& nums) {
        int sptr=nums[nums[0]];
        int fptr= nums[nums[nums[0]]];

        while(sptr!=fptr){
            sptr=nums[sptr];
            fptr= nums[nums[fptr]];

        }

        sptr=nums[0];

        while (sptr!=fptr){
            sptr=nums[sptr];
            fptr=nums[fptr];

        }

        return sptr;
    }
};
